src/views/sms_commands/SmsCommandFormNew.js

```diff
@@ -72,6 +78,23 @@ const useResetFormFields = () => {
     )
 }
 
+// undefined = all, false = program.programType --> WITHOUT_REGISTRATION
+const shouldProgramsBeWithRegistration = parserType => {
+    // EVENT_REGISTRATION uses event programs
+    if (parserType === EVENT_REGISTRATION_PARSER.value) {
+        return PROGRAMS_WITHOUT_REGISTRATION
+    }
+
+    // This is for programs of type tracker,
+    // which can have multiple program stages
+    if (parserType === PROGRAM_STAGE_DATAENTRY_PARSER.value) {
+        return PROGRAMS_WITH_REGISTRATION
+    }
+
+    // Unrelated yet, but in case all program stages are need in the future
+    return ALL_PROGRAMS
+}
```

Comment:
Usually with `should...` type names I'd expect to get back a boolean. Think that something like `getRegistration` or `getProgramType` (or whatever you think is appropriate) would cover what this function does more clearly, to me.

But I'll leave this to you. I assume the top comment (above the function) can be removed?